reference/forms/types/choice.rst

```diff
@@ -95,23 +95,31 @@ method::
             new Category('Cat3'),
             new Category('Cat4'),
         ],
-        'choice_label' => function(Category $category, $key, $value) {
-            return strtoupper($category->getName());
+        // a property path can by used to define the options below
+        'choice_value' => 'name',
+        // the empty value can be passed if there is a placeholder
```

Comment:
I don't understand this comment either:

```
the empty value can be passed if there is a placeholder
```

We're describing the `choice_label` option ... but we talk about "empty value" and "placeholder". I don't see the relation between all this.

Reply:
When a placeholder is set among the choice list it is the empty value (`null` as a choice and empty string as value). When the placeholder is not explicitly defined but the field is not required the component add it as first choice so we can select no choices (opposite of required).
So all callbacks should handle this case, I should better explain that in a dedicated caution I guess.

Reply:
Yes ... maybe a super short comment before the callback that accepts nullable types mentioning that you need that in case you use placeholders ...

... and then maybe a caution message with some more details like you explained in this comment. Thanks!

Reply:
I've pushed a `caution`

Reply:
For the record https://github.com/symfony/symfony/pull/17787